torch/testing/_asserts.py

```diff
@@ -157,7 +185,9 @@ def _check_attributes_equal(
     if check_dtype and actual.dtype != expected.dtype:
         return AssertionError(msg_fmtstr.format("dtype", actual.dtype, expected.dtype))
 
-    if check_stride and actual.stride() != expected.stride():
+    if actual.is_sparse != expected.is_sparse:
```

Comment:
@pearu @cpuhrsch How should this handle different sparse formats?

Reply:
Answered this in https://github.com/pytorch/pytorch/pull/58844#discussion_r642610307